src_features/provideDomainName/cmd_provide_domain_name.c

```diff
@@ -0,0 +1,567 @@
+ * @param[out] domain_name_info the domain name information
+ * @param[] sig_ctx the signature context
+ * @return whether it was successful
+ */
+static bool handle_domain_name(const s_tlv_data *data,
+                               s_domain_name_info *domain_name_info,
+                               s_sig_ctx *sig_ctx) {
+    (void) sig_ctx;
+    strncpy(domain_name_info->name, (const char *) data->value, data->length);
+    domain_name_info->name[data->length] = '\0';
+    return true;
+}
+
+/**
+ * Handler for tag \ref COIN_TYPE
+ *
+ * @param[in] data the tlv data
+ * @param[] domain_name_info the domain name information
+ * @param[] sig_ctx the signature context
+ * @return whether it was successful
+ */
+static bool handle_coin_type(const s_tlv_data *data,
+                             s_domain_name_info *domain_name_info,
+                             s_sig_ctx *sig_ctx) {
+    uint32_t value;
+
+    (void) domain_name_info;
+    (void) sig_ctx;
+    return get_uint_from_data(data, &value) && (value == SLIP_44_ETHEREUM);
+}
+
+/**
+ * Handler for tag \ref ADDRESS
+ *
+ * @param[in] data the tlv data
+ * @param[out] domain_name_info the domain name information
+ * @param[] sig_ctx the signature context
+ * @return whether it was successful
+ */
+static bool handle_address(const s_tlv_data *data,
+                           s_domain_name_info *domain_name_info,
+                           s_sig_ctx *sig_ctx) {
+    (void) sig_ctx;
+    if (data->length != ADDRESS_LENGTH) {
+        return false;
+    }
+    memcpy(domain_name_info->addr, data->value, ADDRESS_LENGTH);
+    return true;
+}
+
+/**
+ * Verify the signature context
+ *
+ * Verify the SHA-256 hash of the payload against the public key
+ *
+ * @param[in] sig_ctx the signature context
+ * @return whether it was successful
+ */
+static bool verify_signature(const s_sig_ctx *sig_ctx) {
+    uint8_t hash[INT256_LENGTH];
+    cx_ecfp_public_key_t verif_key;
+
+    cx_hash((cx_hash_t *) &sig_ctx->hash_ctx, CX_LAST, NULL, 0, hash, INT256_LENGTH);
+    switch (sig_ctx->key_id) {
+#ifdef HAVE_DOMAIN_NAME_TEST_KEY
+        case KEY_ID_TEST:
+#else
+        case KEY_ID_PROD:
+#endif
+            cx_ecfp_init_public_key(CX_CURVE_256K1,
+                                    DOMAIN_NAME_PUB_KEY,
+                                    sizeof(DOMAIN_NAME_PUB_KEY),
+                                    &verif_key);
+            break;
+        default:
+            PRINTF("Error: Unknown metadata key ID %u\n", sig_ctx->key_id);
+            apdu_response_code = APDU_RESPONSE_INVALID_DATA;
+            return false;
+    }
+    if (!cx_ecdsa_verify(&verif_key,
+                         CX_LAST,
+                         CX_SHA256,
+                         hash,
+                         sizeof(hash),
+                         sig_ctx->input_sig,
+                         sig_ctx->input_sig_size)) {
+        PRINTF("Domain name signature verification failed!\n");
+#ifndef HAVE_BYPASS_SIGNATURES
```

Comment:
As is the case in other parts of the app, it's very useful to have these PRINTFs when debugging and can save us a lot of time.
It has no impact on production builds since the [PRINTFs are disabled](https://github.com/LedgerHQ/app-ethereum/blob/7d3bbc703b4595988a24374195df192fdb5b9090/Makefile#L170).

Reply:
I don't mind having a lot of PRINTF at all, I'm a bit more skeptical on leaving debug DEFINES in the code but if you prefer keeping it why not